### PR TITLE
[byutv] Rely on _match_id and _parse_json

### PR DESCRIPTION
**Pull request by TRox1972**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [ ] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [x] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Rely on `_match_id` and `_parse_json`

### CODE REVIEW

File: youtube_dl/extractor/byutv.py [hunk 1]
```diff
@@ -27,15 +26,14 @@ class BYUtvIE(InfoExtractor):
...
 
         webpage = self._download_webpage(url, video_id)
         episode_code = self._search_regex(
             r'(?s)episode:(.*?\}),\s*\n', webpage, 'episode information')
-        episode_json = re.sub(
-            r'(\n\s+)([a-zA-Z]+):\s+\'(.*?)\'', r'\1"\2": "\3"', episode_code)
-        ep = json.loads(episode_json)
+        ep = self._parse_json(re.sub(
+            r'(\n\s+)([a-zA-Z]+):\s+\'(.*?)\'', r'\1"\2": "\3"',
+            episode_code), video_id)
```
Comment (yan12125) [inline youtube_dl/extractor/byutv.py]: ```re.sub``` part can be put in ```transform_source``` parameter of ```_parse_json```.

Comment (TRox1972) [inline youtube_dl/extractor/byutv.py]: done

File: youtube_dl/extractor/byutv.py [hunk 2]
```diff
@@ -27,15 +26,15 @@ class BYUtvIE(InfoExtractor):
...
         webpage = self._download_webpage(url, video_id)
         episode_code = self._search_regex(
             r'(?s)episode:(.*?\}),\s*\n', webpage, 'episode information')
-        episode_json = re.sub(
-            r'(\n\s+)([a-zA-Z]+):\s+\'(.*?)\'', r'\1"\2": "\3"', episode_code)
-        ep = json.loads(episode_json)
+
+        ep = self._parse_json(episode_code, video_id,
+            transform_source=lambda s: re.sub(r'(\n\s+)([a-zA-Z]+):\s+\'(.*?)\'',
+                r'\1"\2": "\3"', s))
```
Comment (dstftw) [inline youtube_dl/extractor/byutv.py]: **Always** check code with flake8.

Comment (TRox1972) [inline youtube_dl/extractor/byutv.py]: Sorry. Fixed now

**Comment by dstftw** [comment]
Cherry-picked in 4da4516973b56bcaa65794a8ae0856cf54740c54, thanks.

**Pull request closed without merging**